nni/nas/evaluator/evaluator.py

```diff
@@ -3,58 +3,337 @@
+
+        1. Use :func:`nni.get_current_parameter` to get the current :class:`~nni.nas.space.ExecutableModelSpace`.
+           Notice that :class:`~nni.nas.space.ExecutableModelSpace` is not a directly-runnable model (e.g., a PyTorch model),
+           which is different from the model received in :meth:`evaluate`.
+           :class:`~nni.nas.space.ExecutableModelSpace` objects are useful for debugging, as well as for some evaluators
+           which need to know extra details about how the model is sampled.
+        2. Use :func:`nni.report_intermediate_result` to report intermediate results.
+        3. Use :func:`nni.report_final_result` to report final results.
+
+        These APIs are only available when the evaluator is executed by NNI,
+        When using evaluator in a standalone way, either use :meth:`trial_available` to check if the APIs are available,
+        or use :meth:`mock_trial` to mock the APIs.
+        Please AVOID using :func:`nni.get_next_parameter()` because NAS framework has already handled the logic
+        of retrieving the next parameter. Incorrectly using :func:`nni.get_next_parameter()` may cause unexpected behavior.
     """
 
-    def evaluate(self, model_cls: Union[Callable[[], Any], Any]) -> Any:
-        """To run evaluation of a model. The model could be either a concrete model or a callable returning a model.
+    def evaluate(self, model: Any) -> Any:
+        """To run evaluation of a model. The model is usually a concrete model.
+        The return value of :meth:`evaluate` can be anything. Typically it's used for test purposes.
 
-        The concrete implementation of evaluate depends on the implementation of ``_execute()`` in sub-class.
+        Subclass should override this.
         """
-        return self._execute(model_cls)
+        raise NotImplementedError()
 
     def __repr__(self):
-        items = ', '.join(['%s=%r' % (k, v) for k, v in self.__dict__.items()])
-        return f'{self.__class__.__name__}({items})'
+        return f'{self.__class__.__name__}({self.extra_repr()})'
+
+    def extra_repr(self) -> str:
+        return ''
+
+    @contextmanager
+    def mock_trial(self, model: ExecutableModelSpace) -> ContextManager[None]:
+        """
+        Context manager to mock trial APIs for standalone usage.
+
+        Under the with-context of this method, :meth:`trial_available` will return ``True``,
+        and :func:`nni.get_current_parameter` will return the given model.
+
+        Parameters
+        ----------
+        model
+            The model to be evaluated. It should be a :class:`~nni.nas.space.ExecutableModelSpace` object.
+
+        Examples
+        --------
+        This method should be mostly used when testing a evaluator.
+        A typical use case is as follows:
+
+        >>> frozen_model = model_space.freeze(sample)
+        >>> with evaluator.mock_trial(frozen_model):
+        ...     evaluator.evaluate(frozen_model.executable_model())
```

Comment:
it seems `frozen_model` is a concrete model, why should we call `executable_model()` to further freeze the model?